qiita_pet/handlers/api_proxy/artifact.py

```diff
@@ -318,21 +319,32 @@ def artifact_post_req(user_id, filepaths, artifact_type, name,
                     if exists['status'] != 'success':
                         return {'status': 'error',
                                 'message': 'File does not exist: %s' % fp}
-                    cleaned_filepaths.append((full_fp, ftype))
+                    if ftype not in cleaned_filepaths:
+                        cleaned_filepaths[ftype] = []
+                    cleaned_filepaths[ftype].append(full_fp)
 
         # This should never happen, but it doesn't hurt to actually have
         # a explicit check, in case there is something odd with the JS
         if not cleaned_filepaths:
             return {'status': 'error',
                     'message': "Can't create artifact, no files provided."}
 
-        job_id = safe_submit(user_id, create_raw_data, artifact_type, prep,
-                             cleaned_filepaths, name=name)
+        command = Command.get_validator(artifact_type)
+        job = ProcessingJob.create(
+            User(user_id),
+            Parameters.load(command, values_dict={
+                'template': prep_template_id,
```

Comment:
`prep_template_id` is a parameter to this function, so there is no new variable created.